design-book/src/roadmap.md

```diff
@@ -0,0 +1,100 @@
+
+The stages contained below are a flexible, coarse resolution roadmap to give newcomers and onlookers a sense of the overall plan and progress.
+If you'd like to add items to this roadmap, open a PR and ping `@alice-i-cecile` as needed for dispute resolution.
+Try to keep each stage small, simple, and easily accomplished.
+We can easily add more stages, but trying to jump ahead to a complex feature risks killing all momentum.
+For actionable, well-scoped items, please open issues instead.
+
+As you complete tasks, check them off here!
+The tasks don't need to be done *well* to be considered completed here: the Bevy community is very good at cleaning up loose strings as we work.
+
+Feel free to tackle any listed item in any order: foundations and primitives can often be built out well before the features that need them are in place.
+This list is not "all of the features that the editor will have": other stages will follow as these features are added.
+It's best not to plan too far in advance!
+
+## Stage 0: Hello Project
+
+- [x] make a repo
+- [x] add a README
+- [x] add a CONTRIBUTING.md
+- [x] add a basic Bevy project
+- [x] define the broad plan for what we're doing
+- [x] define basic constraints for how we're building the editor
+- [x] define a shared vision for what the editor is (and is not)
+
+## Stage 1: Technically an Editor
+
+- [ ] can load scenes from disk using a native file picking widget
+- [ ] can display scenes in a viewer
+  - [ ] 2D
+  - [ ] 3D
+- [ ] simple camera controller
+  - [ ] 2D
+  - [ ] 3D
+- [ ] gizmos can be toggled
+  - [ ] UI outlines
+  - [ ] lights
+  - [ ] AABBs
+- [ ] the list of entities and their components is shown in the inspector
+  - [ ] supports hierarchy via a folding tree view
+- [ ] existing component values can be modified using a graphical interface
+- [ ] resources can be inspected, showing their values and allowing modification
+- [ ] loaded assets can be inspected, providing basic information about them
+- [ ] scenes can be saved back to disk
+
+## Stage 2: Editor-Game Interaction
+
+- [ ] users can press a button in the editor, launching the game from the very beginning
+- [ ] users can press a button in the editor, and their game will run, loading the currently active scene
+- [ ] components can be added and removed, including components specific to the user's game
+- [ ] entities can be spawned and despawned via a GUI
+- [ ] entity hierarchies can be spawned and despawned via a GUI
+  - [ ] manually
+  - [ ] by composing scene files
+- [ ] game-specific rendering techniques display correctly in the editor's scene editing
+- [ ] distinct editor and game views that can be run simultaneously
+
+## Stage 3: UI/UX Foundations
+
+- [ ] tooltips
+- [ ] hotkeys
+- [ ] primitive widgets
+  - [ ] text entry
+  - [ ] numeric entry
+  - [ ] radio buttons
+  - [ ] dropdown
+  - [ ] checkboxes
```

Comment:
I would add a few more very important widgets/primitives just for the game editor:
- Color selector
- File path field (you can change the path to a file via open file dialog instead of manually changing the text)
- Entity field (if a component contains a reference to another entity, Entity field, it would be convenient to change this reference with a convenient simple widget, e.g. selecting an existing entity from a list).
- Image asset previewer